python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -318,13 +318,19 @@ def transform_fn(
         # generators, and in the main event loop, yield them from
         # the queue as they become available.
         output_batch_queue = queue.Queue()
+        exception_queue = queue.Queue()
+        SENTINEL = dict()  # Signal to stop processing
```

Comment:
empty dict can also be a valid UDF return. 
I think we can avoid having this extra exception_queue and SENTINEL. 
Just check if `out_batch = output_batch_queue.get()` is an exception.